tool/site/client-react/tests/components/release/releaseForm.spec.tsx

```diff
@@ -0,0 +1,109 @@
+  expect(releaseTypeRadios).toHaveLength(3)
+  expect(releaseTypeRadios[0].value).toEqual("all")
+  expect(releaseTypeRadios[0].checked).toBeTruthy()
+  expect(releaseTypeRadios[1].value).toEqual("prescriptionId")
+  expect(releaseTypeRadios[1].checked).toBeFalsy()
+  expect(releaseTypeRadios[2].value).toEqual("custom")
+  expect(releaseTypeRadios[2].checked).toBeFalsy()
+
+  const pharmacyContainer = await screen.findByLabelText<HTMLElement>("Pharmacy to release prescriptions to")
+  const pharmacyRadios = pharmacyContainer.getElementsByClassName("nhsuk-radios__input") as HTMLCollectionOf<HTMLInputElement>
+  expect(pharmacyRadios).toHaveLength(3)
+  expect(pharmacyRadios[0].value).toEqual("VNFKT")
+  expect(pharmacyRadios[0].checked).toBeFalsy()
+  expect(pharmacyRadios[1].value).toEqual("YGM1E")
+  expect(pharmacyRadios[1].checked).toBeFalsy()
+  expect(pharmacyRadios[2].value).toEqual("custom")
+  expect(pharmacyRadios[2].checked).toBeFalsy()
+
+  expect(pretty(container.innerHTML)).toMatchSnapshot()
+})
+
+test("Fields default to current values for patient release", async () => {
+  const {container} = render(
+    <ReleaseForm prescriptionId={prescriptionId} onSubmit={jest.fn}/>
+  )
+
+  const releaseTypeContainer = await screen.findByLabelText<HTMLElement>("Choose how you want to release prescription(s)")
+  const releaseTypeRadios = releaseTypeContainer.getElementsByClassName("nhsuk-radios__input") as HTMLCollectionOf<HTMLInputElement>
+  expect(releaseTypeRadios).toHaveLength(3)
+  expect(releaseTypeRadios[0].value).toEqual("all")
+  expect(releaseTypeRadios[0].checked).toBeFalsy()
+  expect(releaseTypeRadios[1].value).toEqual("prescriptionId")
+  expect(releaseTypeRadios[1].checked).toBeTruthy()
+  expect(releaseTypeRadios[2].value).toEqual("custom")
+  expect(releaseTypeRadios[2].checked).toBeFalsy()
+
+  const pharmacyContainer = await screen.findByLabelText<HTMLElement>("Pharmacy to release prescriptions to")
+  const pharmacyRadios = pharmacyContainer.getElementsByClassName("nhsuk-radios__input") as HTMLCollectionOf<HTMLInputElement>
+  expect(pharmacyRadios).toHaveLength(3)
+  expect(pharmacyRadios[0].value).toEqual("VNFKT")
+  expect(pharmacyRadios[0].checked).toBeFalsy()
+  expect(pharmacyRadios[1].value).toEqual("YGM1E")
+  expect(pharmacyRadios[1].checked).toBeFalsy()
+  expect(pharmacyRadios[2].value).toEqual("custom")
+  expect(pharmacyRadios[2].checked).toBeFalsy()
+
+  expect(pretty(container.innerHTML)).toMatchSnapshot()
+})
+
+test("Prescription Id Field is required when releasing by prescription Id", async () => {
+  const {container} = render(
+    <ReleaseForm onSubmit={jest.fn}/>
+  )
+
+  const releaseTypeContainer = await screen.findByLabelText<HTMLElement>("Choose how you want to release prescription(s)")
+  const releaseTypeRadios = releaseTypeContainer.getElementsByClassName("nhsuk-radios__input") as HTMLCollectionOf<HTMLInputElement>
+  userEvent.click(releaseTypeRadios[1])
```

Comment:
Rather than getting by index, is there a way we could use the getByText method instead? Not a huge deal if not, just makes the test easier to read and means it won't break as easily if we rearrange the page / add a new option

Reply:
Will look at in next pr